packages/decoder/lib/index.ts

```diff
@@ -30,76 +30,216 @@ export {
 
 import { Provider } from "web3/providers";
 import { ContractObject } from "@truffle/contract-schema/spec";
+import { ContractConstructorObject, ContractInstanceObject } from "./types";
 
 /**
- * Constructs a contract instance decoder for a given contract instance.
- * @param contract The contract constructor object corresponding to the type of the contract.
- * @param relevantContracts A list of contract constructor objects for other contracts in the project that may be relevant
- * (e.g., providing needed struct or enum definitions, or appearing as a contract type).
- * Including the contract itself here is fine; so is excluding it.
+ * Constructs a wire decoder for the project.
  * @param provider The Web3 provider object to use.
- * @param address The address of the contract instance to decode.  If left out, it will be autodetected on startup.
- *   If an invalid address is provided, this function will throw an exception.
+ * @param contracts A list of contract artifacts for contracts in the project.
+ *
+ *   Contract constructor objects may be substituted for artifacts, so if
+ *   you're not sure which you're dealing with, it's OK.
+ *
+ *   This parameter is intended to be made optional in the future.
  */
-export async function forContractInstance(
-  contract: ContractObject,
-  relevantContracts: ContractObject[],
+export async function forProject(
   provider: Provider,
-  address?: string
-): Promise<ContractInstanceDecoder> {
-  let contractDecoder = await forContract(
-    contract,
-    relevantContracts,
-    provider
-  );
-  let instanceDecoder = await contractDecoder.forInstance(address);
-  return instanceDecoder;
+  contracts: ContractObject[]
+): Promise<WireDecoder> {
+  return new WireDecoder(contracts, provider);
 }
 
 /**
  * Constructs a contract instance decoder for a given contract instance.
- * @param contract The contract constructor object corresponding to the type of the contract.
- * @param relevantContracts A list of contract constructor objects for other contracts in the project that may be relevant
- * (e.g., providing needed struct or enum definitions, or appearing as a contract type).
- * Including the contract itself here is fine; so is excluding it.
+ * @param contract The artifact for the contract.
+ *
+ *   A contract constructor object may be substituted for the artifact, so if
+ *   you're not sure which you're dealing with, it's OK.
  * @param provider The Web3 provider object to use.
+ * @param relevantContracts A list of artifacts for other contracts in the project that may be relevant
```

Comment:
Thought we were just calling all of these `artifacts` instead?

Reply:
Oh.  We could?  That list that I showed you in Slack wasn't intended as the actual variable names, but rather a quick notation for what types the functions took.  (There's more that would need to be changed to be consistent with that, if we really wanted to stick with that.)  I'm for leaving this as-is, personally.

Reply:
Oh I much prefer `artifacts` because it's a name widely-known

Reply:
If you change it only at the surface, it's fine for now, I think

Reply:
OK.  Guess I'll make other changes of `contract` to `artifact` for consistency while I'm at it then, I guess.  (And yeah I have no intention of changing this in the decoders themselves...)